File: backend/app/src/main/java/com/redditclone/app/post/application/PostDownloadDTO.java
package com.redditclone.app.post.application;

import com.redditclone.app.post.domain.PostType;
import lombok.Data;

import java.io.InputStream;
import java.util.UUID;

@Data
public class PostDownloadDTO {
    private UUID id;
    private String title;
    private String content;
    private PostType type;
    private byte[] file;
    private int upvotes;
    private int downvotes;
    private UUID userId;
    private UUID subredditId;
}

File: frontend/src/app/post/dto/post-download.dto.ts
export interface PostDownloadDTO {
  id: string;
  title: string;
  content: string;
  type: PostType;
  file?: string;
  upvotes: number;
  downvotes: number;
  userId: string;
  subredditId: string;
}

export enum PostType {
  TEXT = 'TEXT',
  LINK = 'LINK',
  IMAGE = 'IMAGE',
  VIDEO = 'VIDEO'
}
